core/kazoo_number_manager/src/knm_phone_number.erl

```diff
@@ -486,8 +486,8 @@ set_prev_assigned_to(N, PrevAssignedTo=?MATCH_ACCOUNT_RAW(_)) ->
 -spec used_by(knm_phone_number()) -> api_binary().
 used_by(#knm_phone_number{used_by=UsedBy}) -> UsedBy.
 
--spec set_used_by(knm_phone_number(), ne_binary()) -> knm_phone_number().
-set_used_by(N, UsedBy=?NE_BINARY) ->
+-spec set_used_by(knm_phone_number(), api_binary()) -> knm_phone_number().
+set_used_by(N, UsedBy) ->
```

Comment:
I'd rather you add a clause matching undefined alongside the one matching ?NE_BINARY.
There is a system of checking inputted data and it relies on guards here.
Note: knm_phone_number in #2075 illustrate such matching